other/huffmancoding/huffmancoding_test.go

```diff
@@ -0,0 +1,53 @@
+// huffmancoding_test.go
+// description: Tests the compression, encoding and decoding algorithms of huffmancoding.go.
+// author(s) [pedromsrocha](https://github.com/pedromsrocha)
+// see huffmancoding.go
+
+package huffman
```

Comment:
This is an algorithm that is quite involved for people unfamilar to this (I myself took quite some time to review this thoroughly), so I recomment creating blackbox testing by changing the package name to
```suggestion
package huffman_test
```

That way, people who want to understand the algorithm from a high level perspective, it would be easier, and it would also ensure that there is nothing in code that has leaky abstraction :smile:

Reply:
Done with 6cd8d3a. I rename the functions of the Huffman API to avoid future clashes as new compression methods are added to the package.